src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    super(name, health, hidden);
```

Comment:
[CODE QUALITY] The constructor of the Carnivore class should not include a `hidden` parameter, as it is not a property of the base Animal class, nor is it used in the Carnivore class itself. You should remove the `hidden` parameter to avoid confusion and potential bugs.